### PR TITLE
Convert named_colors example to use Rectangle

### PR DESCRIPTION
**Pull request by asongtoruin**
Converts the named colours example to use `Rectangle` rather than `hlines` for ease of future updating. Discussions still ongoing in #15614 as to whether the colour swatches should be given borders or not.

### CODE REVIEW

File: examples/color/named_colors.py [hunk 1]
```diff
@@ -59,15 +60,16 @@ def plot_colortable(colors, title, sort_colors=True, emptycols=0):
...
 
         ax.text(text_pos_x, y, name, fontsize=14,
                 horizontalalignment='left',
                 verticalalignment='center')
 
-        ax.hlines(y, swatch_start_x, swatch_end_x,
-                  color=colors[name], linewidth=18)
+        ax.add_patch(
+            Rectangle(xy=(swatch_start_x, y-9), width=swatch_width,
+                      height=18, fill=True, facecolor=colors[name])
```
Comment (timhoffm) [inline examples/color/named_colors.py]:
```suggestion
                      height=18, facecolor=colors[name])
```
Filling is default for patches, no need to specify this explicitly.

**Comment by story645** [comment]
Is the plan to open a new PR for the borders?

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to matplotlib. We'd love to see you back in the future.

Note: I've squashed your commits before merging. We try to keep commits to logical units to keep the history more readable. You can use `git commit -a --amend --no-edit` and then `git push origin [mybranch] --force` to update a PR without adding additional commits.

**Comment by asongtoruin** [comment]
@timhoffm apologies - I just accepted through the GH interface, I'll push it through manually next time.

@story645 there didn't seem to be a consensus as to whether we should change it or not, but it's a fairly trivial change I can make for a new PR if you feel like further discussion would be helpful?

**Comment by story645** [comment]
Yes, if you don't mind please make a new PR for the border.

**Pull request merged**